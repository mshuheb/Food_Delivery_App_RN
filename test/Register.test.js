import React from "react";
import { render, fireEvent, waitFor } from "@testing-library/react-native";
import { Alert } from "react-native"; // Ensure Alert is imported to be mocked
import Register from "../app/(authenticate)/register";
import { supabase } from "../supabase";
import { useRouter } from "expo-router";

// Mock external modules
jest.mock("../supabase");
jest.mock("expo-router", () => ({
  useRouter: jest.fn(),
}));

// Mock expo-vector-icons
jest.mock("@expo/vector-icons", () => ({
  AntDesign: "AntDesign", // Mocking icon
  MaterialIcons: "MaterialIcons", // Mocking icon
  Ionicons: "Ionicons", // Mocking icon
}));

// Mock Alert
jest.spyOn(Alert, "alert").mockImplementation(() => {});

describe("<Register />", () => {
  let routerMock;

  beforeEach(() => {
    routerMock = { replace: jest.fn() };
    useRouter.mockReturnValue(routerMock);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test("renders correctly with initial UI elements", () => {
    const { getByPlaceholderText, getByText, toJSON } = render(<Register />);

    expect(getByText("Food App")).toBeTruthy();
    expect(getByText("Register to your account")).toBeTruthy();
    expect(getByPlaceholderText("Enter your Name")).toBeTruthy();
    expect(getByPlaceholderText("Enter your Email")).toBeTruthy();
    expect(getByPlaceholderText("Enter your password")).toBeTruthy();
    expect(getByText("Select your Gender:")).toBeTruthy();
    expect(getByText("Male")).toBeTruthy();
    expect(getByText("Female")).toBeTruthy();
    expect(getByText("Register")).toBeTruthy();
    expect(getByText("Already have an Account? Sign In")).toBeTruthy();

    // Snapshot test
    expect(toJSON()).toMatchSnapshot();
  });

  test("shows alert on registration error", async () => {
    supabase.auth.signUp.mockResolvedValue({
      error: { message: "Email already exists" },
      data: null,
    });

    const { getByText, getByPlaceholderText, toJSON } = render(<Register />);

    const nameInput = getByPlaceholderText("Enter your Name");
    const emailInput = getByPlaceholderText("Enter your Email");
    const passwordInput = getByPlaceholderText("Enter your password");
    const registerButton = getByText("Register");

    fireEvent.changeText(nameInput, "John Doe");
    fireEvent.changeText(emailInput, "test@example.com");
    fireEvent.changeText(passwordInput, "password123");
    fireEvent.press(registerButton);

    await waitFor(() => {
      expect(supabase.auth.signUp).toHaveBeenCalledWith({
        email: "test@example.com",
        password: "password123",
      });
      expect(Alert.alert).toHaveBeenCalledWith("Error while registering", "Please try again");
    });

    // Snapshot test
    expect(toJSON()).toMatchSnapshot();
  });

  test("navigates to login screen after successful registration", async () => {
    supabase.auth.signUp.mockResolvedValue({
      error: null,
      data: { user: { role: "authenticated" } },
    });

    supabase.from.mockReturnValue({
      insert: jest.fn().mockResolvedValue({ error: null }),
    });

    const { getByText, getByPlaceholderText, toJSON } = render(<Register />);
    const nameInput = getByPlaceholderText("Enter your Name");
    const emailInput = getByPlaceholderText("Enter your Email");
    const passwordInput = getByPlaceholderText("Enter your password");
    const registerButton = getByText("Register");

    fireEvent.changeText(nameInput, "John Doe");
    fireEvent.changeText(emailInput, "test@example.com");
    fireEvent.changeText(passwordInput, "password123");
    fireEvent.press(registerButton);

    await waitFor(() => {
      expect(supabase.auth.signUp).toHaveBeenCalledWith({
        email: "test@example.com",
        password: "password123",
      });
      expect(supabase.from).toHaveBeenCalledWith("user_details");
      expect(Alert.alert).toHaveBeenCalledWith(
        "Success",
        "You have been successfully registered. Please check your email for confirmation."
      );
      expect(routerMock.replace).toHaveBeenCalledWith("/login");
    });

    // Snapshot test
    expect(toJSON()).toMatchSnapshot();
  });

  test("redirects to login screen on 'Sign In' press", () => {
    const { getByText, toJSON } = render(<Register />);
    const signInText = getByText("Already have an Account? Sign In");

    fireEvent.press(signInText);

    expect(routerMock.replace).toHaveBeenCalledWith("/login");

    // Snapshot test
    expect(toJSON()).toMatchSnapshot();
  });
});
