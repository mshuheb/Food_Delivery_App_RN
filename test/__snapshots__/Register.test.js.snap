// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Register /> navigates to login screen after successful registration 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "marginTop": 50,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Food App
    </Text>
  </View>
  <View
    onLayout={[Function]}
  >
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <Text
        style={
          {
            "color": "red",
            "fontSize": 17,
            "fontWeight": "bold",
            "marginTop": 12,
          }
        }
      >
        Register to your account
      </Text>
    </View>
    <View
      style={
        {
          "marginTop": 70,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#E0E0E0",
            "borderRadius": 5,
            "flexDirection": "row",
            "gap": 5,
            "marginTop": 30,
            "paddingVertical": 5,
          }
        }
      >
        <Ionicons
          color="gray"
          name="person"
          size={24}
          style={
            {
              "marginLeft": 8,
            }
          }
        />
        <TextInput
          onChangeText={[Function]}
          placeholder="Enter your Name"
          style={
            {
              "color": "gray",
              "marginVertical": 10,
              "width": 300,
            }
          }
          value="John Doe"
        />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#E0E0E0",
            "borderRadius": 5,
            "flexDirection": "row",
            "gap": 5,
            "marginTop": 30,
            "paddingVertical": 5,
          }
        }
      >
        <MaterialIcons
          color="gray"
          name="email"
          size={24}
          style={
            {
              "marginLeft": 8,
            }
          }
        />
        <TextInput
          onChangeText={[Function]}
          placeholder="Enter your Email"
          style={
            {
              "color": "gray",
              "marginVertical": 10,
              "width": 300,
            }
          }
          value="test@example.com"
        />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#E0E0E0",
            "borderRadius": 5,
            "flexDirection": "row",
            "gap": 5,
            "marginTop": 30,
            "paddingVertical": 5,
          }
        }
      >
        <AntDesign
          color="black"
          name="lock1"
          size={24}
          style={
            {
              "marginLeft": 8,
            }
          }
        />
        <TextInput
          onChangeText={[Function]}
          placeholder="Enter your password"
          secureTextEntry={true}
          style={
            {
              "color": "gray",
              "marginVertical": 10,
              "width": 300,
            }
          }
          value="password123"
        />
      </View>
      <View
        style={
          {
            "marginTop": 30,
          }
        }
      >
        <Text
          style={
            {
              "color": "gray",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Select your Gender:
        </Text>
        <View
          style={
            {
              "flexDirection": "row",
              "marginTop": 10,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderRadius": 50,
                "borderWidth": 1,
                "flexDirection": "row",
                "marginRight": 20,
                "paddingHorizontal": 10,
                "paddingVertical": 5,
              }
            }
          >
            <Ionicons
              color="gray"
              name="male"
              size={24}
            />
            <Text
              style={
                {
                  "color": "gray",
                  "marginLeft": 5,
                }
              }
            >
              Male
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderRadius": 50,
                "borderWidth": 1,
                "flexDirection": "row",
                "paddingHorizontal": 10,
                "paddingVertical": 5,
              }
            }
          >
            <Ionicons
              color="gray"
              name="female"
              size={24}
            />
            <Text
              style={
                {
                  "color": "gray",
                  "marginLeft": 5,
                }
              }
            >
              Female
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#fd5c63",
          "borderRadius": 6,
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": 50,
          "padding": 15,
          "width": 200,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Register
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "marginTop": 15,
        }
      }
    >
      <Text
        style={
          {
            "color": "gray",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Already have an Account? Sign In
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`<Register /> redirects to login screen on 'Sign In' press 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "marginTop": 50,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Food App
    </Text>
  </View>
  <View
    onLayout={[Function]}
  >
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <Text
        style={
          {
            "color": "red",
            "fontSize": 17,
            "fontWeight": "bold",
            "marginTop": 12,
          }
        }
      >
        Register to your account
      </Text>
    </View>
    <View
      style={
        {
          "marginTop": 70,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#E0E0E0",
            "borderRadius": 5,
            "flexDirection": "row",
            "gap": 5,
            "marginTop": 30,
            "paddingVertical": 5,
          }
        }
      >
        <Ionicons
          color="gray"
          name="person"
          size={24}
          style={
            {
              "marginLeft": 8,
            }
          }
        />
        <TextInput
          onChangeText={[Function]}
          placeholder="Enter your Name"
          style={
            {
              "color": "gray",
              "marginVertical": 10,
              "width": 300,
            }
          }
          value=""
        />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#E0E0E0",
            "borderRadius": 5,
            "flexDirection": "row",
            "gap": 5,
            "marginTop": 30,
            "paddingVertical": 5,
          }
        }
      >
        <MaterialIcons
          color="gray"
          name="email"
          size={24}
          style={
            {
              "marginLeft": 8,
            }
          }
        />
        <TextInput
          onChangeText={[Function]}
          placeholder="Enter your Email"
          style={
            {
              "color": "gray",
              "marginVertical": 10,
              "width": 300,
            }
          }
          value=""
        />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#E0E0E0",
            "borderRadius": 5,
            "flexDirection": "row",
            "gap": 5,
            "marginTop": 30,
            "paddingVertical": 5,
          }
        }
      >
        <AntDesign
          color="black"
          name="lock1"
          size={24}
          style={
            {
              "marginLeft": 8,
            }
          }
        />
        <TextInput
          onChangeText={[Function]}
          placeholder="Enter your password"
          secureTextEntry={true}
          style={
            {
              "color": "gray",
              "marginVertical": 10,
              "width": 300,
            }
          }
          value=""
        />
      </View>
      <View
        style={
          {
            "marginTop": 30,
          }
        }
      >
        <Text
          style={
            {
              "color": "gray",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Select your Gender:
        </Text>
        <View
          style={
            {
              "flexDirection": "row",
              "marginTop": 10,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderRadius": 50,
                "borderWidth": 1,
                "flexDirection": "row",
                "marginRight": 20,
                "paddingHorizontal": 10,
                "paddingVertical": 5,
              }
            }
          >
            <Ionicons
              color="gray"
              name="male"
              size={24}
            />
            <Text
              style={
                {
                  "color": "gray",
                  "marginLeft": 5,
                }
              }
            >
              Male
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderRadius": 50,
                "borderWidth": 1,
                "flexDirection": "row",
                "paddingHorizontal": 10,
                "paddingVertical": 5,
              }
            }
          >
            <Ionicons
              color="gray"
              name="female"
              size={24}
            />
            <Text
              style={
                {
                  "color": "gray",
                  "marginLeft": 5,
                }
              }
            >
              Female
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#fd5c63",
          "borderRadius": 6,
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": 50,
          "padding": 15,
          "width": 200,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Register
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "marginTop": 15,
        }
      }
    >
      <Text
        style={
          {
            "color": "gray",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Already have an Account? Sign In
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`<Register /> renders correctly with initial UI elements 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "marginTop": 50,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Food App
    </Text>
  </View>
  <View
    onLayout={[Function]}
  >
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <Text
        style={
          {
            "color": "red",
            "fontSize": 17,
            "fontWeight": "bold",
            "marginTop": 12,
          }
        }
      >
        Register to your account
      </Text>
    </View>
    <View
      style={
        {
          "marginTop": 70,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#E0E0E0",
            "borderRadius": 5,
            "flexDirection": "row",
            "gap": 5,
            "marginTop": 30,
            "paddingVertical": 5,
          }
        }
      >
        <Ionicons
          color="gray"
          name="person"
          size={24}
          style={
            {
              "marginLeft": 8,
            }
          }
        />
        <TextInput
          onChangeText={[Function]}
          placeholder="Enter your Name"
          style={
            {
              "color": "gray",
              "marginVertical": 10,
              "width": 300,
            }
          }
          value=""
        />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#E0E0E0",
            "borderRadius": 5,
            "flexDirection": "row",
            "gap": 5,
            "marginTop": 30,
            "paddingVertical": 5,
          }
        }
      >
        <MaterialIcons
          color="gray"
          name="email"
          size={24}
          style={
            {
              "marginLeft": 8,
            }
          }
        />
        <TextInput
          onChangeText={[Function]}
          placeholder="Enter your Email"
          style={
            {
              "color": "gray",
              "marginVertical": 10,
              "width": 300,
            }
          }
          value=""
        />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#E0E0E0",
            "borderRadius": 5,
            "flexDirection": "row",
            "gap": 5,
            "marginTop": 30,
            "paddingVertical": 5,
          }
        }
      >
        <AntDesign
          color="black"
          name="lock1"
          size={24}
          style={
            {
              "marginLeft": 8,
            }
          }
        />
        <TextInput
          onChangeText={[Function]}
          placeholder="Enter your password"
          secureTextEntry={true}
          style={
            {
              "color": "gray",
              "marginVertical": 10,
              "width": 300,
            }
          }
          value=""
        />
      </View>
      <View
        style={
          {
            "marginTop": 30,
          }
        }
      >
        <Text
          style={
            {
              "color": "gray",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Select your Gender:
        </Text>
        <View
          style={
            {
              "flexDirection": "row",
              "marginTop": 10,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderRadius": 50,
                "borderWidth": 1,
                "flexDirection": "row",
                "marginRight": 20,
                "paddingHorizontal": 10,
                "paddingVertical": 5,
              }
            }
          >
            <Ionicons
              color="gray"
              name="male"
              size={24}
            />
            <Text
              style={
                {
                  "color": "gray",
                  "marginLeft": 5,
                }
              }
            >
              Male
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderRadius": 50,
                "borderWidth": 1,
                "flexDirection": "row",
                "paddingHorizontal": 10,
                "paddingVertical": 5,
              }
            }
          >
            <Ionicons
              color="gray"
              name="female"
              size={24}
            />
            <Text
              style={
                {
                  "color": "gray",
                  "marginLeft": 5,
                }
              }
            >
              Female
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#fd5c63",
          "borderRadius": 6,
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": 50,
          "padding": 15,
          "width": 200,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Register
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "marginTop": 15,
        }
      }
    >
      <Text
        style={
          {
            "color": "gray",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Already have an Account? Sign In
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`<Register /> shows alert on registration error 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "marginTop": 50,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Food App
    </Text>
  </View>
  <View
    onLayout={[Function]}
  >
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <Text
        style={
          {
            "color": "red",
            "fontSize": 17,
            "fontWeight": "bold",
            "marginTop": 12,
          }
        }
      >
        Register to your account
      </Text>
    </View>
    <View
      style={
        {
          "marginTop": 70,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#E0E0E0",
            "borderRadius": 5,
            "flexDirection": "row",
            "gap": 5,
            "marginTop": 30,
            "paddingVertical": 5,
          }
        }
      >
        <Ionicons
          color="gray"
          name="person"
          size={24}
          style={
            {
              "marginLeft": 8,
            }
          }
        />
        <TextInput
          onChangeText={[Function]}
          placeholder="Enter your Name"
          style={
            {
              "color": "gray",
              "marginVertical": 10,
              "width": 300,
            }
          }
          value="John Doe"
        />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#E0E0E0",
            "borderRadius": 5,
            "flexDirection": "row",
            "gap": 5,
            "marginTop": 30,
            "paddingVertical": 5,
          }
        }
      >
        <MaterialIcons
          color="gray"
          name="email"
          size={24}
          style={
            {
              "marginLeft": 8,
            }
          }
        />
        <TextInput
          onChangeText={[Function]}
          placeholder="Enter your Email"
          style={
            {
              "color": "gray",
              "marginVertical": 10,
              "width": 300,
            }
          }
          value="test@example.com"
        />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#E0E0E0",
            "borderRadius": 5,
            "flexDirection": "row",
            "gap": 5,
            "marginTop": 30,
            "paddingVertical": 5,
          }
        }
      >
        <AntDesign
          color="black"
          name="lock1"
          size={24}
          style={
            {
              "marginLeft": 8,
            }
          }
        />
        <TextInput
          onChangeText={[Function]}
          placeholder="Enter your password"
          secureTextEntry={true}
          style={
            {
              "color": "gray",
              "marginVertical": 10,
              "width": 300,
            }
          }
          value="password123"
        />
      </View>
      <View
        style={
          {
            "marginTop": 30,
          }
        }
      >
        <Text
          style={
            {
              "color": "gray",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Select your Gender:
        </Text>
        <View
          style={
            {
              "flexDirection": "row",
              "marginTop": 10,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderRadius": 50,
                "borderWidth": 1,
                "flexDirection": "row",
                "marginRight": 20,
                "paddingHorizontal": 10,
                "paddingVertical": 5,
              }
            }
          >
            <Ionicons
              color="gray"
              name="male"
              size={24}
            />
            <Text
              style={
                {
                  "color": "gray",
                  "marginLeft": 5,
                }
              }
            >
              Male
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderRadius": 50,
                "borderWidth": 1,
                "flexDirection": "row",
                "paddingHorizontal": 10,
                "paddingVertical": 5,
              }
            }
          >
            <Ionicons
              color="gray"
              name="female"
              size={24}
            />
            <Text
              style={
                {
                  "color": "gray",
                  "marginLeft": 5,
                }
              }
            >
              Female
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#fd5c63",
          "borderRadius": 6,
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": 50,
          "padding": 15,
          "width": 200,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Register
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "marginTop": 15,
        }
      }
    >
      <Text
        style={
          {
            "color": "gray",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Already have an Account? Sign In
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
